src/controller/p2p/preheat/controllor_test.go

```diff
@@ -295,3 +317,18 @@ func (s *preheatSuite) TestCheckHealth() {
 	err = s.controller.CheckHealth(s.ctx, instance)
 	s.NoError(err)
 }
+
+type fakePreheatScheduler struct {
```

Comment:
There is a mock scheduler under `src/testing/pkg/scheduler`

Reply:
Please use that one for mock cases

Reply:
changed